tests/core/test_utils.py

```diff
@@ -254,3 +259,8 @@ def helper(fn, arg, shared_list):
         p.join()
 
     assert results.count(True) == 1, f"Failed for path: {path}"
+
+
+def test_get_aws_region_from_processing_job_arn():
+    test_arn = "arn:aws:sagemaker:us-east-1:012345678910:processing-job/random-test-arn"
+    assert get_aws_region_from_processing_job_arn(test_arn) == "us-east-1"
```

Comment:
Might require refactoring but can we add a test that verifies the request itself to be sent instead of just the region?